src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -769,4 +809,13 @@ private static void fillModulesFromCheckstylePackage() {
         NAME_TO_FULL_MODULE_NAME.put("Checker", BASE_PACKAGE + ".Checker");
         NAME_TO_FULL_MODULE_NAME.put("TreeWalker", BASE_PACKAGE + ".TreeWalker");
     }
+
+    /**
+     * Sets the field useBruteForceModuleLoading.
+     * @param useBruteForceModuleLoading the new value.
+     */
+    public void setUseBruteForceModuleLoading(boolean useBruteForceModuleLoading) {
```

Comment:
design on setters is bad design. User should not have ability to change this option in runtime, so it is defined at the beginning and last for end of run time.
Please put this option to c-tor. 
As you make it c-tor option, TC will complain about boolean usage in public method, with unclear meaning. It will force you to create Enum with options {Discover, DiscoverWithFallbackToTryInEachPackage}. Discover is default.

Reply:
done